backend/src/main/java/com/github/lenaruppert/backend/controller/MongoUserController.java

```diff
@@ -0,0 +1,102 @@
+package com.github.lenaruppert.backend.controller;
+
+import com.github.lenaruppert.backend.model.MongoUser;
+import com.github.lenaruppert.backend.model.MongoUserDTO;
+import com.github.lenaruppert.backend.repository.MongoUserRepository;
+import jakarta.servlet.http.HttpSession;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.security.Principal;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/user")
+@RequiredArgsConstructor
+public class MongoUserController {
```

Comment:
Generally I would advise to put all the logic into the Service. Like that you have only the endpoints in the controller that call then the service. But this is a matter of personal taste I think :)